docs/user/gcp/install_upi.md

```diff
@@ -0,0 +1,589 @@
+in the initial manifests within the Ignition configuration. To use the output
+of the default, generated `ignition-configs` extracting the internal
+infrastructure name is necessary.
+
+An example of a way to get this is below:
+
+```console
+$ jq -r .infraID metadata.json
+openshift-vw9j6
+```
+
+Export variables to be used in examples below.
+
+```sh
+export BASE_DOMAIN='example.com'
+export BASE_DOMAIN_ZONE_NAME='example'
+export NETWORK_CIDR='10.0.0.0/16'
+export MASTER_SUBNET_CIDR='10.0.0.0/19'
+export WORKER_SUBNET_CIDR='10.0.32.0/19'
+
+export CLUSTER_NAME=`jq -r .clusterName metadata.json`
+export INFRA_ID=`jq -r .infraID metadata.json`
+export PROJECT_NAME=`jq -r .gcp.projectID metadata.json`
+export REGION=`jq -r .gcp.region metadata.json`
+```
+## Create the VPC
+
+Copy the [01_vpc.py](../../../upi/gcp/01_vpc.py) locally.
+
+Create a resource definition file: `01_vpc.yaml`
+
+```console
+$ cat <<EOF >01_vpc.yaml
+imports:
+- path: 01_vpc.py
+
+resources:
+- name: cluster-vpc
+  type: 01_vpc.py
+  properties:
+    infra_id: '${INFRA_ID}'
+    region: '${REGION}'
+
+    master_subnet_cidr: '${MASTER_SUBNET_CIDR}'
+    worker_subnet_cidr: '${WORKER_SUBNET_CIDR}'
+EOF
+```
+- infra_id: the infra_id from above (ie openshift-xxxx)
+- region: the region to deploy the cluster into (ie us-east1)
+- master_subnet_cidr: the cidr for the master subnet (ie 10.0.0.0/19)
+- worker_subnet_cidr: the cidr for the worker subnet (ie 10.0.32.0/19)
+
+Create the deployment using gcloud.
+
+```sh
+gcloud deployment-manager deployments create ${INFRA_ID}-vpc --config 01_vpc.yaml
+```
+
+## Create DNS entries and load balancers
+
+Copy the [02_infra.py](../../../upi/gcp/02_infra.py) locally.
+
+Export variables needed by the resource definition.
+
+```sh
+export CLUSTER_NETWORK=`gcloud compute networks describe ${INFRA_ID}-network --format json | jq -r .selfLink`
+```
+
+Create a resource definition file: `02_infra.yaml`
+
+```console
+$ cat <<EOF >02_infra.yaml
+imports:
+- path: 02_infra.py
+
+resources:
+- name: cluster-infra
+  type: 02_infra.py
+  properties:
+    infra_id: '${INFRA_ID}'
+    region: '${REGION}'
+
+    cluster_domain: '${CLUSTER_NAME}.${BASE_DOMAIN}'
+    cluster_network: '${CLUSTER_NETWORK}'
+EOF
+```
+- infra_id: the infra_id from above (ie openshift-xxxx)
+- region: the region to deploy the cluster into (ie us-east1)
+- cluster_domain: the domain for the cluster (ie openshift.example.com)
+- cluster_network: the url to the cluster network object
+
+Create the deployment using gcloud.
+
+```sh
+gcloud deployment-manager deployments create ${INFRA_ID}-infra --config 02_infra.yaml
+```
+
+The templates do not create DNS entries due to limitations of Deployment
+Manager, so we must create them manually.
+
+```sh
+export CLUSTER_IP=`gcloud compute addresses describe ${INFRA_ID}-cluster-public-ip --region=${REGION} --format json | jq -r .address`
+
+# Add external DNS entries
+if [ -f transaction.yaml ]; then rm transaction.yaml; fi
```

Comment:
I don't now if we need this first guard in our docs (although it's fine if in local scripts if folks want to script out this whole doc flow).  For these docs, we can probably just assume we're running in a clean directory.  And you can have something like:

    rm transaction.yaml  # remove the transaction file after a successful execution

after the `execute` call.

Reply:
This reduces the collisions if they are following by hand and have to re-do this step or something similar.